internal/handlers/pg/pgdb/indexes.go

```diff
@@ -55,43 +56,79 @@ func Indexes(ctx context.Context, tx pgx.Tx, db, collection string) ([]Index, er
 		res[i] = idx.Index
 	}
 
-	// TODO Add tests that indexes sorted correctly: https://github.com/FerretDB/FerretDB/issues/1509
-	slices.SortFunc(res, func(a, b Index) bool { return a.Name < b.Name })
-
 	return res, nil
 }
 
-// createIndex creates a new index for the given params.
-// TODO This method will become exported in https://github.com/FerretDB/FerretDB/issues/1509.
-func createIndex(ctx context.Context, tx pgx.Tx, db, collection string, i *Index) error {
+// CreateIndexIfNotExists creates a new index for the given params if such an index doesn't exist.
+//
+// If the index exists, it doesn't return an error.
+// If the collection doesn't exist, it will be created and then the index will be created.
+func CreateIndexIfNotExists(ctx context.Context, tx pgx.Tx, db, collection string, i *Index) error {
+	if err := CreateCollectionIfNotExists(ctx, tx, db, collection); err != nil {
+		return err
+	}
+
 	pgTable, pgIndex, err := newMetadataStorage(tx, db, collection).setIndex(ctx, i.Name, i.Key, i.Unique)
 	if err != nil {
 		return err
 	}
 
-	if err := createPgIndexIfNotExists(ctx, tx, db, pgTable, pgIndex, true); err != nil {
+	if err := createPgIndexIfNotExists(ctx, tx, db, pgTable, pgIndex, i.Key, i.Unique); err != nil {
 		return err
 	}
 
 	return nil
 }
 
 // createPgIndexIfNotExists creates a new index for the given params if it does not exist.
-func createPgIndexIfNotExists(ctx context.Context, tx pgx.Tx, schema, table, index string, isUnique bool) error {
+func createPgIndexIfNotExists(ctx context.Context, tx pgx.Tx, schema, table, index string, fields IndexKey, isUnique bool) error {
+	if len(fields) == 0 {
+		return lazyerrors.Errorf("no fields for index")
+	}
+
 	var err error
 
 	unique := ""
 	if isUnique {
 		unique = " UNIQUE"
 	}
 
+	fieldsDef := make([]string, len(fields))
+
+	for i, field := range fields {
+		var order string
+
+		switch field.Order {
+		case types.Ascending:
+			order = "ASC"
+		case types.Descending:
+			order = "DESC"
+		default:
+			return lazyerrors.Errorf("unknown sort order: %d", field.Order)
+		}
+
+		// It's important to sanitize field.Field data here, as it's a user-provided value.
+		fieldsDef[i] = fmt.Sprintf(`((_jsonb->%s)) %s`, quoteString(field.Field), order)
```

Comment:
Why instead of creating new function can't we just use SQL arguments here?

Reply:
@noisersup could you suggest what exactly you mean? My idea with a separate function was to highlight how exactly we sanitize the user input. The `QuoteString` function used to be an exported function in the `pgx` package a while ago (no it's in the `internal` repo, so it's not really exported to the other repos), so my idea was to show that use case.

Reply:
Can't we just

```suggestion
		// It's important to sanitize field.Field data here, as it's a user-provided value.
		fieldsDef[i] = fmt.Sprintf(`((_jsonb->%s)) %s`, p.Next(), order)
		args := []any{field.Field}
```
So we get something like: `((_jsonb->$1))`?

Reply:
The root cause is that DDL statements like `CREATE INDEX` do not accept normal arguments

Reply:
That's interesting

Reply:
It seems that this function might be unsafe when specific options on pg are set [see](https://github.com/jackc/pgx/issues/868#issuecomment-725544647)

Can we verify if this affects us, and maybe add a check for those config variables?

Reply:
That's an interesting finding.

I have a test case `"dangerous-key"` that contains SQL injection. If sanitization is not set, the injection passes, and the test fails.  Do you think it's enough? Or should there be some extra checks?

Reply:
(upd: found where to add the check! thanks a lot @noisersup!)

Reply:
Ok, I added a check to `pool.go - pgPool. checkConnection`. I wanted to add is somewhere here too, but that would require us to keep `pgPool` here and in many other places (e.g. like `createCollection`), which will lead to having `pgPool` everywhere.

@noisersup @AlekSi does the current check look enough to you?